### PR TITLE
Dont double count data sent

### PR DESCRIPTION
**Pull request by dirkmc**
Fixes https://github.com/filecoin-project/go-data-transfer/issues/184

Tasks:
- [x] Implement fix
- [x] Unit tests
- [x] Integration tests

### CODE REVIEW

File: impl/events.go [hunk 1]
```diff
@@ -28,63 +28,101 @@ func (m *manager) OnChannelOpened(chid datatransfer.ChannelID) error {
...
-				return errors.New("stop processing")
-			}
-			return nil
-		})
-		if err != nil || result != nil {
-			msg, err := m.processRevalidationResult(chid, result, err)
-			if msg != nil {
-				if err := m.dataTransferNetwork.SendMessage(context.TODO(), chid.Initiator, msg); err != nil {
-					return err
-				}
```
Comment (dirkmc) [inline impl/events.go:54]: Note: the logic hasn't changed here, just the indentation

File: channels/channels_test.go [hunk 2]
```diff
@@ -145,46 +145,52 @@ func TestChannels(t *testing.T) {
...
+		require.True(t, isNew)
 		state = checkEvent(ctx, t, received, datatransfer.DataSent)
 		require.Equal(t, uint64(50), state.Received())
 		require.Equal(t, uint64(100), state.Sent())
 		require.Equal(t, []cid.Cid{cids[0]}, state.ReceivedCids())
 
 		// errors if channel does not exist
-		err = channelList.DataReceived(datatransfer.ChannelID{Initiator: peers[1], Responder: peers[0], ID: tid1}, cids[1], 200)
+		isNew, err = channelList.DataReceived(datatransfer.ChannelID{Initiator: peers[1], Responder: peers[0], ID: tid1}, cids[1], 200)
 		require.True(t, xerrors.As(err, new(*channels.ErrNotFound)))
```
Comment (aarshkshah1992) [inline channels/channels_test.go:169]: assert `isNew` is false when we get an error

File: channels/channels_test.go [hunk 3]
```diff
@@ -145,46 +145,52 @@ func TestChannels(t *testing.T) {
...
-		err = channelList.DataSent(datatransfer.ChannelID{Initiator: peers[1], Responder: peers[0], ID: tid1}, cids[1], 200)
+		isNew, err = channelList.DataSent(datatransfer.ChannelID{Initiator: peers[1], Responder: peers[0], ID: tid1}, cids[1], 200)
```
Comment (aarshkshah1992) [inline channels/channels_test.go:170]: assert false.

File: channels/channels.go [hunk 4]
```diff
@@ -357,31 +360,32 @@ func (c *Channels) removeSeenCIDCaches(chid datatransfer.ChannelID) error {
...
-// onProgress fires an event indicating progress has been made in
-// queuing / sending / receiving blocks.
-// These events are fired only for new blocks (not for example if
-// a block is resent)
-func (c *Channels) fireProgressEvent(chid datatransfer.ChannelID, evt datatransfer.EventCode, progressEvt datatransfer.EventCode, k cid.Cid, delta uint64) error {
+// fireProgressEvent fires
+// - an event for queuing / sending / receiving blocks
+// - a corresponding "progress" event if the block has not been seen before
+// For example if a block is resent, the method will fire DataSent but not DataSentProgress.
+// Returns true if a progress event was fired.
```
Comment (aarshkshah1992) [inline channels/channels.go:367]: However, if the block is being sent for the first time, the method will fire both DataSent AND DataSentProgress.

**Review by aarshkshah1992 — APPROVED** [review]
@dirkmc  LGTM. Just some nits we need to fix in the test.

**Pull request merged**